src/messaging/ExchangeMgr.cpp

```diff
@@ -213,29 +213,33 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
         msgFlags.Set(MessageFlagValues::kDuplicateMessage);
     }
 
-    // Search for an existing exchange that the message applies to. If a match is found...
-    bool found = false;
-    mContextPool.ForEachActiveObject([&](auto * ec) {
-        if (ec->MatchExchange(session, packetHeader, payloadHeader))
-        {
-            // Found a matching exchange. Set flag for correct subsequent MRP
-            // retransmission timeout selection.
-            if (!ec->HasRcvdMsgFromPeer())
+    // Skip retrieval of exchange for group message
+    if (!session.GetGroupId().HasValue())
```

Comment:
We could, but wouldn't this be handled automatically by MatchExchange returning false?  As in, is this correctness fix or optimization?  If optimization, please document that.

In any case, this should also use `IsGroupSession`.

Reply:
OK, please document that clearly in the code.